### PR TITLE
Themes Showcase: Add beta badge to current theme card

### PR DESCRIPTION
**Pull request by jeyip**
### Changes proposed in this Pull Request
Add a Beta label to the current theme banner for block-based themes (tagged with block-templates) in the theme showcase. Also removes customized beta badge styling on the FSE tab to ensure consistent sizing between all beta badges.

#### Before
Screen Shot 2021-09-08 at 5 25 19 PM

#### After
Screen Shot 2021-09-15 at 12 59 33 PM

### Testing instructions

- Open Appearance -> Themes
- Confirm current themes that support block-templates have a Beta badge

Related to https://github.com/Automattic/wp-calypso/issues/54508

### CODE REVIEW

**Comment by jeyip** [comment]
**Note:**

For me, it feels like the number of beta badges might be making the themes showcase a bit visually crowded. With the changes in this PR, we'd see badges in three different locations.

Screen Shot 2021-09-08 at 5 28 09 PM

Wondering if others feel the same way, and if we do, then what we think the best path forward might be. cc @onuro @vindl

**Comment by onuro** [comment]
> For me, it feels like the number of beta badges might be making the themes showcase a bit visually crowded. With the changes in this PR, we'd see badges in three different locations.

@jeyip I agree with you, however trying to resolve that could bring some confusions in the user's end. Removing it or altering it's style of it's instances may be interpreted as they're not FSE. Open for any suggestions though.

**Comment by vindl** [comment]
> Wondering if others feel the same way, and if we do, then what we think the best path forward might be. cc @onuro @vindl

I don't have a strong opinion about this. While I agree with you that it's a bit crowded, users also tend to glance over the details in the cards (especially on busy screens) so making sure that they don't miss them is also a factor.

I don't think it's particularly important to get this perfect right now. To me it's sufficient to get something working to start the beta rollout, as we will almost inevitably iterate on whatever we come up with based on user feedback and testing.

**Comment by jeyip** [comment]
> I don't think it's particularly important to get this perfect right now. To me it's sufficient to get something working to start the beta rollout

This sounds good to me. 

> Removing it or altering it's style of it's instances may be interpreted as they're not FSE. Open for any suggestions though.

Why don't I make a non-blocking follow-up issue for the #cylon backlog? Then we can move forward with this PR.

File: client/utils.js [hunk 1]
```diff
@@ -7,3 +7,8 @@ export function pathToRegExp( path ) {
 	}
 	return new RegExp( '^' + path + '(/.*)?$' );
 }
+
+export function isFullSiteEditingTheme( theme ) {
+	const features = theme?.taxonomies?.theme_feature;
+	return features && features.some( ( feature ) => feature.slug === 'block-templates' );
+}
```
Comment (Addison-Stavlo) [inline client/utils.js]: I think we might also need to check beta eligibility/status before adding the badges?

That is, soon we will roll out Universal themes to all users allowing non-beta users to activate them and use as the standard/classic experience.  We wouldn't want the beta badge showing up on these Universal themes in that situation. 🤔

**Review by Addison-Stavlo — COMMENTED** [review]
Tested and it works well, but I think we might need to consider the use of universal themes by non-beta users in whether or not these badges are added.

File: client/utils.js (hunk 1 above)
Comment (jeyip) [reply client/utils.js]: I'm curious how you feel about using the `wpcom_is_fse_activated` option to disable / enable the beta badge instead, as opposed to the `core-site-editor-enabled` sticker.

I added a guard to evaluate `isSiteUsingCoreSiteEditor` selector, which references the `is_core_site_editor_enabled` attribute. The attribute is fetched from the rest API which is resolved from the `wpcom_is_fse_activated` option. 

The beta badge should only be displayed on a core FSE enabled site. One thing to note, however, is that using the `wpcom_is_fse_activated` option means that if a user enrolled in the beta disables the Appearance > Site Editor "Enable Site Editor" setting, all beta badges will disappear on Universal / block-based themes.

This makes sense to me because the user would then be opted into the standard/classic experience of Universal Themes, but before updating the other PRs, would love to hear your thoughts.

Comment (Addison-Stavlo) [reply client/utils.js]: Im not entirely certain either way.

>  if a user enrolled in the beta disables the Appearance > Site Editor "Enable Site Editor" setting, all beta badges will disappear on Universal / block-based themes.

At this point, that toggle isnt available to users.  And whether or not we add a toggle for users to do this sort of depends on the p2 discussion @creativecoder brought up recently.

Whether or not it makes sense for that to remove the beta tag also depends on where/how we end up having this sort of toggle.  If it is a prompt on theme switch, then those themes probably still need the beta tag to convey to the user that they can get back to the beta experience by activating them.  If that toggle is elsewhere, like at the top of the showcase itself, in the 'beta testing' menu item, or in Appearance->Site Editor, etc. then it seems like it would be ok to hide the 'beta' tags on the theme cards when that toggle is off. 🤔 

TLDR - I think it depends on what/how we set up that toggle and present it to the user.

Comment (jeyip) [reply client/utils.js]: > At this point, that toggle isnt available to users. And whether or not we add a toggle for users to do this sort of depends on the p2 discussion @creativecoder brought up recently.
>
> Whether or not it makes sense for that to remove the beta tag also depends on where/how we end up having this sort of toggle.

Gotcha -- I mistakenly thought the toggle would be made available to users in Appearance->Site Editor upon release of the beta. Thanks for the info Addie 👍

Comment (creativecoder) [reply client/utils.js]: Great question to ask! I think tying the Beta badge to the `core-site-editor-enabled` blog sticker makes the most sense to me.

The main reason being that once the beta is over, we'll likely keep the `wpcom_is_fse_activated` option and related functionality (but probably default it to on) so that users can still have a "classic" theme experience, if they want. But I imagine we'll remove all of the visual indications of "Beta", and ignore the `core-site-editor-enabled` sticker at the same time, once that's considered the default functionality for new sites.

File: client/utils.js [hunk 2]
```diff
@@ -7,3 +7,8 @@ export function pathToRegExp( path ) {
 	}
 	return new RegExp( '^' + path + '(/.*)?$' );
 }
+
+export function isFullSiteEditingTheme( theme ) {
```
Comment (Copons) [inline client/utils.js]: I've left a comment about this in #56225

https://github.com/Automattic/wp-calypso/pull/56225/files/4076b2ddf9e69c65c7ec40b55e110dfbd7fefbcd#r717716762

**Review by Copons — APPROVED** [review]
This works well in my tests! ✨ 

One thing I've noticed, which is not related to this change but is exacerbated by the new beta badge, is that we should probably polish the `CurrentTheme`'s responsive behaviours a little bit.
See how the overlaps and overflows at different (narrow) widths:
Screenshot 2021-10-01 at 15 30 21
Screenshot 2021-10-01 at 15 30 31

**Review by vindl — APPROVED** [review]
LGTM! :shipit:

**Comment by jeyip** [comment]
With the existing changes in this PR, beta badges will be shown when the `wpcom_is_fse_activated` option is enabled. Ideally, it would be displayed, instead, when a site has been flagged with a `core-fse-eligible` blog sticker.

After brief discussions with @vindl, I'll go ahead and merge what we have for now since efforts to expose information about the fse eligibility sticker are still underway (D67076-code) The current changes won't cause any defects, circling back and accommodating the sticker changes won't require any major refactoring, and on top of that, we'll finally get this work codified into production.

**Comment by jeyip** [comment]
> One thing I've noticed, which is not related to this change but is exacerbated by the new beta badge, is that we should probably polish the CurrentTheme's responsive behaviours a little bit.

I'll make a follow-up issue to track this shortly.

**Comment by jeyip** [comment]
> I'll make a follow-up issue to track this shortly.

https://github.com/Automattic/wp-calypso/issues/56853
